patches/tModLoader/Terraria/ModLoader/AccessorySlotLoader.cs

```diff
@@ -0,0 +1,536 @@
+		}
+
+		/// <summary>
+		/// Mirrors Player.UpdateDyes() for modded slots
+		/// </summary>
+		public void VanillaUpdateDyesMirror(Player player) {
+			var modSlotPlayer = player.GetModPlayer<ModAccessorySlotPlayer>();
+			for (int i = 0; i < modSlotPlayer.exAccessorySlot.Length; i++) {
+				if (ModdedIsAValidEquipmentSlotForIteration(i)) {
+					int num = i % modSlotPlayer.exDyesAccessory.Length;
+					player.UpdateItemDye(i < modSlotPlayer.exDyesAccessory.Length, modSlotPlayer.exHideAccessory[num], modSlotPlayer.exAccessorySlot[i], modSlotPlayer.exDyesAccessory[num]);
+				}
+			}
+		}
+
+		/// <summary>
+		/// Mirrors Player.GetPreferredGolfBallToUse.
+		/// Provides the golf ball projectile from an accessory slot. 
+		/// </summary>
+		public bool VanillaPreferredGolfBallMirror(ref int projType) {
+			for (int num = list.Count * 2 - 1; num >= 0; num--) {
+				if (ModdedIsAValidEquipmentSlotForIteration(num)) {
+					Item item2 = ModSlotPlayer.exAccessorySlot[num];
+					if (!item2.IsAir && item2.shoot > 0 && ProjectileID.Sets.IsAGolfBall[item2.shoot]) {
+						projType = item2.shoot;
+						return true;
+					}
+				}
+			}
+			return false;
+		}
+
+		/// <summary>
+		/// Mirrors the ItemSlot.DyeSwap method; 
+		/// Unfortunately, I (Solxan) couldn't ever get ItemSlot.DyeSwap invoked so pretty sure this and its vanilla code is defunct.
+		/// Here in case someone proves my statement wrong later.
+		/// </summary>
+		public Item VanillaDyeSwapMirror(Item item, out bool success) {
+			Item item2 = item;
+			int dyeSlotCount = list.Count;
+
+			for (int i = 0; i < list.Count; i++) {
+				if (ModSlotPlayer.exDyesAccessory[i].type == 0) {
+					dyeSlotCount = i;
+					break;
+				}
+			}
+
+			if (dyeSlotCount >= list.Count) {
+				success = false;
+				return item2;
+			}
+
+			item2 = ModSlotPlayer.exDyesAccessory[dyeSlotCount].Clone();
+			ModSlotPlayer.exDyesAccessory[dyeSlotCount] = item.Clone();
+
+			SoundEngine.PlaySound(7);
+			Recipe.FindRecipes();
+			success = true;
+			return item2;
+		}
+
+
+		/// <summary>
+		/// Mirrors Player.LoadPlayer_LastMinuteFixes. Really Vanilla?
+		/// Corrects the player.lavaMax time, wingsLogic, and no fall dmg to be accurate
+		/// </summary>
+		public void VanillaLastMinuteFixesMirror(Player newPlayer) {
+			ModAccessorySlotPlayer ModSlotPlayer = newPlayer.GetModPlayer<ModAccessorySlotPlayer>();
+			for (int i = 0; i < list.Count; i++) {
+				int type = ModSlotPlayer.exAccessorySlot[i].type;
+				if (type == 908 || type == 4874 || type == 5000)
+					newPlayer.lavaMax += 420;
+
+				if (type == 906 || type == 4038)
+					newPlayer.lavaMax += 420;
+
+				if (newPlayer.wingsLogic == 0 && ModSlotPlayer.exAccessorySlot[i].wingSlot >= 0) {
+					newPlayer.wingsLogic = ModSlotPlayer.exAccessorySlot[i].wingSlot;
+					newPlayer.equippedWings = ModSlotPlayer.exAccessorySlot[i];
+				}
+				
+				if (type == 158 || type == 396 || type == 1250 || type == 1251 || type == 1252)
+					newPlayer.noFallDmg = true;
```

Comment:
Rather than duplicating the vanilla code here, it would be ideal if the approach used for the rest of the equipment effects was followed. By which I mean:

Split the vanilla `LastMinuteFixes` method into one that takes an `Item` parameter, and the vanilla accessory slot iteration just calls that one. That way mods can call it too.

Then have a `LastMinuteFixes` hook on `ModPlayer` which you use to apply your custom accessory slots.